python/cuml/common/array.py

```diff
@@ -196,7 +196,7 @@ def __cuda_array_interface__(self):
         return output
 
     @with_cupy_rmm
-    def to_output(self, output_type='cupy'):
+    def to_output(self, output_type='cupy', output_dtype=None):
```

Comment:
I think a lot of the confitionals for constructors of the type `if output_type` could be avoided if you do 

```python
if output_type is None:
    output_type=self.dtype
```
which would allow common constructors if I'm not mistaken

Reply:
Good call. Current logic doesn't assume underlying data structure conversion always avoids the copy if the dtypes are the same. This should be unnecessary

Reply:
Should be resolved by https://github.com/rapidsai/cuml/pull/2431/commits/4f46c731c03ced34a8bf6210d58c72c6166ca2e7